lua/luasnip/config.lua

```diff
@@ -214,29 +215,54 @@ c = {
 	end,
 
 	_setup = function()
-		vim.cmd(
-			string.format(
-				[[
-        augroup luasnip
-            au!
-            autocmd %s * lua require("luasnip").unlink_current_if_deleted()
-            autocmd %s * lua require("luasnip").active_update_dependents()
-            autocmd %s * lua require("luasnip").exit_out_of_region(require("luasnip").session.current_nodes[vim.api.nvim_get_current_buf()])
-			"Remove buffers' nodes on deletion+wipeout.
-			autocmd BufDelete,BufWipeout * lua current_nodes = require("luasnip").session.current_nodes if current_nodes then current_nodes[tonumber(vim.fn.expand("<abuf>"))] = nil end
-		]]
-					.. (session.config.enable_autosnippets and [[
-			autocmd InsertCharPre * lua Luasnip_just_inserted = true
-			autocmd TextChangedI,TextChangedP * lua if Luasnip_just_inserted then require("luasnip").expand_auto() Luasnip_just_inserted=nil end
-		]] or "")
-					.. [[
-		augroup END
-		]],
-				session.config.delete_check_events,
-				session.config.update_events,
-				session.config.region_check_events
-			)
-		)
+		local augroup = vim.api.nvim_create_augroup("luasnip", {})
+		vim.api.nvim_create_autocmd(session.config.delete_check_events, {
+			callback = function()
+				require("luasnip").unlink_current_if_deleted()
+			end,
+			group = augroup,
+		})
+		vim.api.nvim_create_autocmd(session.config.update_events, {
+			callback = function()
+				require("luasnip").active_update_dependents()
+			end,
+			group = augroup,
+		})
+		vim.api.nvim_create_autocmd(session.config.region_check_events, {
+			callback = function()
+				require("luasnip").exit_out_of_region(
+					require("luasnip").session.current_nodes[vim.api.nvim_get_current_buf()]
+				)
+			end,
+			group = augroup,
+		})
+		-- Remove buffers' nodes on deletion+wipeout.
+		vim.api.nvim_create_autocmd({ "BufDelete", "BufWipeout" }, {
+			callback = function()
+				current_nodes = require("luasnip").session.current_nodes
```

Comment:
@L3MON4D3  I think this should be `local` (which is wrong in the original as well)

Reply:
tru I also thought that
but I wasn't sure if there is a global created somewhere else as well